src/ekat/util/ekat_math_utils.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef EKAT_MATH_UTILS_HPP
+#define EKAT_MATH_UTILS_HPP
+
+#include "ekat/ekat.hpp"
+
+#include "Kokkos_Core.hpp"
+
+#ifndef KOKKOS_ENABLE_CUDA
+# include <cmath>
+# include <algorithm>
+#endif
+
+namespace ekat {
+namespace util {
+
+#ifdef KOKKOS_ENABLE_CUDA
+// Replacements for namespace std functions that don't run on the GPU.
+template <typename T>
+KOKKOS_INLINE_FUNCTION
+const T& min (const T& a, const T& b) {
+  return a < b ? a : b;
+}
+
+template <typename T>
+KOKKOS_INLINE_FUNCTION
+const T& max (const T& a, const T& b) {
+  return a > b ? a : b;
+}
+
+template <typename T>
+KOKKOS_INLINE_FUNCTION
+bool isfinite (const T& a) {
```

Comment:
is_finite, maybe?

Reply:
Ah, I wanted to change this too. The problem is that the STL has `isfinite`. A few lines below, you can see that if the exec space is not cuda we simply do `using std::isfinite`. In order to have `is_finite`, we'd have to impl a wrapper fcn around `std::isfinite`. It's doable, maybe a bit tedious. Do you think the benefit of snake case outweigh the benefit of using the same fcn name as the STL? I don't have a strong opinion either way.

Reply:
I forgot about the standard function. Yeah, no need to be over-consistent with snake case (C++ certainly isn't!). I think it's fine as is.